crates/witness-generator/src/eest_generator.rs

```diff
@@ -0,0 +1,196 @@
+
+        let mut cmd = Command::new("./scripts/download-and-extract-fixtures.sh");
+        if let Some(tag) = tag {
+            cmd.arg(tag);
+        }
+        let output = cmd.output()?;
+
+        if !output.status.success() {
+            bail!(
+                "Failed to download EEST benchmark fixtures: {}",
+                String::from_utf8_lossy(&output.stderr)
+            );
+        }
+
+        Ok(ExecSpecTestBlocksAndWitnesses {
+            directory_path: PathBuf::from_str("./zkevm-fixtures")?,
+            include,
+            exclude,
+        })
+    }
+}
+
+/// Witness generator that produces `BlocksAndWitnesses` for EEST fixtures.
+#[derive(Debug, Clone)]
+pub struct ExecSpecTestBlocksAndWitnesses {
+    directory_path: PathBuf,
+    include: Vec<String>,
+    exclude: Vec<String>,
+}
+
+impl Drop for ExecSpecTestBlocksAndWitnesses {
+    fn drop(&mut self) {
+        if self.directory_path.exists() {
+            match std::fs::remove_dir_all(&self.directory_path) {
+                Ok(_) => {}
+                Err(e) => eprintln!(
+                    "Failed to remove directory {}: {}",
+                    self.directory_path.display(),
+                    e
+                ),
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl WitnessGenerator for ExecSpecTestBlocksAndWitnesses {
+    // Generates blocks and witnesses from the EEST fixtures located in the specified directory,
+    // filtering by the provided include and exclude patterns.
+    async fn generate(&self) -> Result<Vec<BlocksAndWitnesses>> {
+        let suite_path = self.directory_path.join("fixtures/blockchain_tests");
+
+        if !suite_path.exists() {
+            bail!("Test suite path does not exist: {}.", suite_path.display());
+        }
+
+        let test_file_paths = find_all_files_with_extension(&suite_path, ".json");
+        let mut tests: Vec<(String, BlockchainTest)> = Vec::new();
+        for path in test_file_paths {
+            let test_case = match BlockchainTestCase::load(&path) {
+                Ok(case) => case,
+                Err(e) => {
+                    eprintln!("Failed to load test case from {}: {e}", path.display());
+                    continue;
+                }
+            };
+
+            let file_tests: Vec<(String, BlockchainTest)> = test_case
+                .tests
+                .into_iter()
+                .map(|(name, case)| {
+                    (
+                        name.split('/').next_back().unwrap_or(&name).to_string(),
+                        case,
+                    )
+                })
+                .filter(|(name, _)| !self.exclude.iter().any(|filter| name.contains(filter)))
+                .filter(|(name, _)| self.include.iter().all(|f| name.contains(f)))
+                .map(|(name, case)| (name, case))
+                .collect();
+            tests.extend(file_tests);
+        }
+
+        let bws: Result<Vec<_>> = tests
+            .par_iter()
```

Comment:
Parallelized generation. This wasn't the case before.

Reply:
Ah, it's a bit heavy in `/tmp` usage of tmp folders... maybe you experienced some problem reg filled `/tmp` or something.

If this is annoying we can remove it